cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -42,6 +43,41 @@ bool gdv_fn_like_utf8_utf8(int64_t ptr, const char* data, int data_len,
   return (*holder)(std::string(data, data_len));
 }
 
+const char* gdv_fn_regexp_replace_utf8_utf8(
+    int64_t ptr, int64_t holder_ptr, const char* data, int32_t data_len,
+    const char* /*pattern*/, int32_t /*pattern_len*/, const char* replace_string,
+    int32_t replace_string_len, int32_t* out_length) {
+  std::string user_input(data, data_len);
+  std::string replace_input(replace_string, replace_string_len);
+  gandiva::ExecutionContext* context = reinterpret_cast<gandiva::ExecutionContext*>(ptr);
+
+  gandiva::ReplaceHolder* holder = reinterpret_cast<gandiva::ReplaceHolder*>(holder_ptr);
+  bool was_replaced = (*holder)(context, user_input, replace_input, out_length);
+
+  if (!was_replaced) {
+    return data;
+  }
+
+  // This condition treats the case where the whole string is replaced by an empty string
+  if (*out_length == 0) {
+    return "";
+  }
+
+  char* result_buffer =
+      reinterpret_cast<char*>(gdv_fn_context_arena_malloc(ptr, *out_length));
+
+  if (result_buffer == nullptr) {
+    std::string err_msg = "Could not allocate memory for string: " + user_input;
+    gdv_fn_context_set_error_msg(ptr, err_msg.data());
+    *out_length = 0;
+    return "";
+  }
+
+  memcpy(result_buffer, user_input.c_str(), *out_length);
```

Comment:
Use user_input.data() instead of c_str() (Latter I believe does another allocation and copy)